### ISSUE TITLE
[Bug]: mongodbatlas_team removes member either the change doesn't show it

### ISSUE THREAD

**Issue by lolejar** [mongodb/terraform-provider-mongodbatlas#2662]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Provider Version

1.20.0

### Terraform Version

1.9.7

### Terraform Edition

Terraform Open Source (OSS)

### Current Behavior

This started happening since Thursday 03.10.2024 without anychanges to IaC.
I have a IaC then runs each 10 minutes (for more over than 2 years) that synchronizes access to MongoDB teams from external management system.

When I run terraform apply / plan it shows me change in usernames as expected. When I terraform apply, usernames are added to state, however they are not being created in MongoDB Team AND the execution of provider fails with message it was not possible to remove user as it CANNOT_REMOVE_LAST_USER_FROM_TEAM. 

That user it tries to remove is the one that I am aware cannot be removed and is included in default_team_members and is always member. To me it looks like Terraform wants to remove all usernames and add them again. Plan shows only additions, no removals.

### Terraform configuration to reproduce the issue

```hcl
resource "mongodbatlas_team" "managers" {
  for_each  = local.mongodb_projects__map
  org_id    = var.organization_id
  name      = "${each.value.project_name} - Managers"
  usernames = concat(sort(module.default_team_members.emails), sort(module.managers_emails["${each.value.resource}"].emails))

  depends_on = [
    mongodbatlas_org_invitation.organization_owners,
    mongodbatlas_org_invitation.organization_members,
    mongodbatlas_org_invitation.organization_readers
  ]
}
```

### Steps To Reproduce

1. I run apply
2. Users for teams are fetched from external system and their emails are being resolved
3. Terraform provider detect changes between state and MongoDB
4. It shows only additions to the team (1 team member always present - 1 unchanged element hidden )
5. It starts applying changes 
6. It fails that is cannon delete that 1 team member always present
7. Remaining users are added to state
8. Remaining users are not added to MongoDB Team
9. Only member of MongoDB Team is default team member

### Logs

```txt
# mongodbatlas_team.managers["team_key"] will be updated in-place
  ~ resource "mongodbatlas_team" "managers" {
        id        = "<redacted>"
        name      = "<redacted>"
      ~ usernames = [
          + "xxx@aaa.com",
          + "yyy@aaa.com",
          + "zzz@aaa.com",
            # (1 unchanged element hidden) 
        ]
        # (2 unchanged attributes hidden)
    }

mongodbatlas_team.managers["team_key"]: Modifying... [id=<redacted>]
╷
│ Error: error deleting Atlas User (<team>) information: https://cloud.mongodb.com/api/atlas/v2/orgs/<org-id>/teams/<team-id>/users/<default team member> DELETE: HTTP 400 Bad Request (Error code: "CANNOT_REMOVE_LAST_USER_FROM_TEAM") Detail: A team must contain at least one user. Reason: Bad Request. Params: []
│ 
│   with mongodbatlas_team.managers["team_key"],
│   on project.tf line 86, in resource "mongodbatlas_team" "managers":
│   86: resource "mongodbatlas_team" "managers" {
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by lolejar** [comment]
I just tested it with completely new resource and it was created succesfully 

```hcl
resource "mongodbatlas_team" "test" {
  org_id    = var.organization_id
  name      = "Test"
  usernames = [ "aaa@xxx.com"]
}
```

when i updated resource 

```hcl
resource "mongodbatlas_team" "test" {
  org_id    = var.organization_id
  name      = "Test"
  usernames = [ "aaa@xxx.com", "yyy@xxx.com" ]
}
```

```txt
  # mongodbatlas_team.test will be updated in-place
  ~ resource "mongodbatlas_team" "test" {
        id        = "<redacted>"
        name      = "Test"
      ~ usernames = [
          + "yyy@xxx.com",
            # (1 unchanged element hidden)
        ]
        # (2 unchanged attributes hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

 Error: error deleting Atlas User (<team-id>) information: https://cloud.mongodb.com/api/atlas/v2/orgs/<org-id>/teams/<team-id>/users/<aaa@xxx.com user id> DELETE: HTTP 400 Bad Request (Error code: "CANNOT_REMOVE_LAST_USER_FROM_TEAM") Detail: A team must contain at least one user. Reason: Bad Request. Params: []
│ 
│   with mongodbatlas_team.test,
│   on project.tf line 112, in resource "mongodbatlas_team" "test":
│  112: resource "mongodbatlas_team" "test" {
```

**Comment by AgustinBettati** [comment]
Hello @lolejar, thank you for raising this issue.
I was able to reproduce the issue and can see it is related to a new validation in the underlying API. Will keep you updated on any changes from API or terraform provider side.

**Comment by MD-t2024** [comment]
Same issue here too with terraform "terraform_version": "1.0.11",

The job was running fine last week, however did a change to add/remove users this week and start getting this error in all such team modules :

Error: error deleting Atlas User () information: DELETE https://cloud.mongodb.com/api/atlas/v1.0/orgs/<org-id>/teams/<team-id>/users: 400 (request "CANNOT_REMOVE_LAST_USER_FROM_TEAM") A team must contain at least one user.
│

**Comment by AgustinBettati** [comment]
Hi @lolejar @MD-t2024, the issue has been fixed and released as part of [v1.21.1](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CHANGELOG.md#1211-october-09-2024).

Additionally we are looking into remediation options from the API side to ensure existing terraform provider versions are not impacted. Will leave a comment when I have more updates.

Closing the issue but feel free to reopen or comment if there are remaining doubts.
